lib/formalism/form/coercion.rb

```diff
@@ -75,21 +75,19 @@ def to_float
 		end
 
 		def to_time
-			return if @value.nil?
-
-			Time.parse(@value)
+			Time.parse(@value) if @value && !@value.empty?
 		end
 
 		def to_boolean
 			@value && @value.to_s != 'false' ? true : false
 		end
 
 		def to_symbol
-			@value&.to_sym
+			@value.to_sym if @value.respond_to?(:to_sym)
```

Comment:
I'm not sure that this is OK, not exception. There are two ways (options): suppress or raise. I think raising is OK until we have problems with this.

Reply:
I'm not sure that forms should ever raise when valid from form data or json are parsed

I understand that validation errors will not be created here, but raise seems to be too brutal.
If you aren't sure too, i think better is to use my version

Reply:
> when valid from form data

I don't think that something without `to_sym` method for `Symbol` field is valid.

> or json

It will have `to_sym` method.

Reply:
I mean… 

You can't cover any type for any (business) case.

`Integer` doesn't have `to_sym` method, but theoretically it can be converted.

You don't know should it be converted or not for some case. Let users to decide.

Reply:
> You don't know should it be converted or not for some case. Let users to decide.

It is a good idea, but how? By handling `NoMethodErrors`?

Reply:
> I don't think that something without to_sym method for Symbol field is valid.

I mean valid as `application/x-www-form-urlencoded`, not valid as data for specific form.

Reply:
> I don't think that something without to_sym method for Symbol field is valid.

Validation collects errors to a Set, not raises something

Reply:
> It is a good idea, but how? By handling NoMethodErrors?

Maybe. Or fix the input. Or option for Formalism in the future if needed. We don't need for this now.

Reply:
Ok, i will remove corresponding tests and revert changes

Reply:
Done